tests/aeroval/test_setupclasses.py

```diff
@@ -1,44 +1,211 @@
+from copy import deepcopy
+
 import pytest
 
 from pyaerocom.aeroval.setupclasses import EvalSetup
 from pyaerocom.exceptions import EvalEntryNameError
-from tests.fixtures.aeroval.cfg_test_exp1 import CFG
+from tests.fixtures.aeroval.cfg_test_exp1 import CFG, MODELS, OBS_GROUNDBASED
 
 
-@pytest.mark.parametrize(
-    "cfg,update",
-    [
-        ("cfgexp1", dict()),
-        ("cfgexp1", dict(obs_cfg=dict(OBS=CFG["obs_cfg"]["AERONET-Sun"]))),
-    ],
-)
-def test_EvalSetup___init__(eval_config: dict, update: dict):
-    eval_config.update(update)
-    EvalSetup(**eval_config)
+@pytest.fixture
+def cfg_exp1(update: dict | None) -> dict:
+    cfg = deepcopy(CFG)
+    if update:
+        cfg.update(update)
+
+    return cfg
+
+
+@pytest.fixture
+def eval_setup(cfg_exp1: dict) -> EvalSetup:
+    return EvalSetup.model_validate(cfg_exp1)
+
+
+default_config = pytest.mark.parametrize("update", (pytest.param(None, id="default"),))
+
+
+@default_config
+def test_EvalSetup(cfg_exp1: dict):
+    assert EvalSetup(**cfg_exp1) == EvalSetup.model_validate(cfg_exp1)
 
 
 @pytest.mark.parametrize(
-    "cfg,update,error",
+    "update,error",
     [
-        (
-            "cfgexp1",
-            dict(model_cfg=dict(WRONG_MODEL=CFG["model_cfg"]["TM5-AP3-CTRL"])),
+        pytest.param(
+            dict(model_cfg=dict(WRONG_MODEL=MODELS["TM5-AP3-CTRL"])),
             "Invalid name: WRONG_MODEL",
+            id="model_cfg",
         ),
-        (
-            "cfgexp1",
-            dict(obs_cfg=dict(WRONG_OBS=CFG["obs_cfg"]["AERONET-Sun"])),
+        pytest.param(
+            dict(obs_cfg=dict(WRONG_OBS=OBS_GROUNDBASED["AERONET-Sun"])),
             "Invalid name: WRONG_OBS",
+            id="obs_cfg",
         ),
-        (
-            "cfgexp1",
+        pytest.param(
             dict(obs_cfg=dict(OBS=dict(web_interface_name="WRONG_OBS"))),
             "Invalid name: WRONG_OBS",
+            id="web_interface_name",
         ),
     ],
 )
-def test_EvalSetup___init__INVALID_ENTRY_NAMES(eval_config: dict, update: dict, error: str):
-    eval_config.update(update)
+def test_EvalSetup_INVALID_ENTRY_NAMES(cfg_exp1: dict, error: str):
     with pytest.raises(EvalEntryNameError) as e:
-        EvalSetup(**eval_config)
+        EvalSetup(**cfg_exp1)
     assert error in str(e.value)
+
+    with pytest.raises(EvalEntryNameError) as e:
+        EvalSetup.model_validate(cfg_exp1)
+    assert error in str(e.value)
+
+
+@default_config
+def test_EvalSetup_ProjectInfo(eval_setup: EvalSetup, cfg_exp1: dict):
```

Comment:
The tests for `ProjectInfo`,  `ExperimentInfo`, and `TimeSetup` look like they're just checking what given in the default `CFG`, correct? IIRC, there are many overlaps between the default settings and what appears in the `CFG`, which caused issues in the previous tests where although these fields were not being set correctly, the tests did not fail. Are you sure these tests are robust enough to prevent against that happening again?

Reply:
I can add parametrized test for those to test overriding the defaults.

Reply:
Should be fixed now